### DISCUSSION TITLE
Proposal: CUE modules storage model

### DISCUSSION THREAD

**Discussion by rogpeppe** [cue-lang/cue#2449] [Proposal]
Proposal #2330 presented a updated modules proposal. This discussion is for a sub-proposal of that: details on how we propose to store modules in OCI registries.

See here for the sub-proposal text in full:
https://github.com/cue-lang/proposal/blob/main/designs/modules/2449-modules-storage-model.md

**Comment by mxey** [comment]
I looked over the proposal and this would work for us.

**Comment by extemporalgenome** [comment]
In terms of format, zstd and brotli are often used to supersede gzip, as both offer better average compression ratios.

I'm not too familiar with brotli, but zstd does also have faster compression/decompression speed as well.

**Reply by myitcv** [reply]
Thanks @extemporalgenome - @mvdan brought this up recently too. It's on the list of points to consider

**Reply by mvdan** [reply]
And it is being separately tracked as https://github.com/cue-lang/cue/issues/2744, since it's a decision to be made :)

**Reply by myitcv** [reply]
Totally forgot I'd created that!

**Comment by myitcv** [comment]
This proposal was superseded by #2939. As we updated in https://github.com/cue-lang/cue/discussions/2939#discussioncomment-9468945, that proposal (and its sub-proposals) have now been accepted, hence we are closing this discussion to represent the decision on the v3 proposal.
